Assets/Scripts/Controllers/PlayerController.cs

```diff
@@ -15,23 +15,51 @@ void Awake()
     
     public HealthBar healthBar;
     public ManaBar manaBar;
-    
+    [SerializeField]public int currentMana;
+    [SerializeField]public int currentLevel;    
     private LayerMask NPCMask;
     private int maxMana = 10;
-    private int currentMana;
     private IInteractable interactable;
+    public bool saved;
+
 
     protected new void Start()
     {
-        UIManager.FinalRoomScore = 6;
-
-        NPCMask = LayerMask.GetMask("NPC");
-        name = "Player";
-        maxHealth = 15;
-        base.Start();
-        healthBar.SetMaxHealth(maxHealth);
-        currentMana = maxMana;
-        manaBar.SetMaxMana(maxMana);
+        saved = GameStartManager.PlayingSavedGame;
+        Debug.Log("SAVED IS "+saved);
+
+        if (saved)
```

Comment:
There's some duplicate lines of code in each clause of this if statement (ex. NPCMask = LayerMask.GetMask("NPC");). These should be pulled outside the if statement.